probe-rs/src/architecture/arm/ap/mod.rs

```diff
@@ -13,32 +13,44 @@ pub use memory_ap::{
 use super::Register;
 use thiserror::Error;
 
-#[derive(Debug, Error, PartialEq)]
+#[derive(Debug, Error)]
 pub enum AccessPortError {
-    #[error("Invalid Access PortType Number")]
-    InvalidAccessPortNumber,
-    #[error("Misaligned memory access")]
-    MemoryNotAligned,
+    #[error("Failed to access address 0x{0:08x} as it is not aligned to 4 bytes.")]
+    MemoryNotAligned(u32),
     #[error("Failed to read register {name}, address 0x{address:08x}")]
-    RegisterReadError { address: u8, name: &'static str },
+    RegisterReadError {
+        address: u8,
+        name: &'static str,
+        inner: Box<dyn std::error::Error + Send + Sync>,
```

Comment:
We should use either `#[from]` or `#[source]` here, to ensure the `Error::source` method is correctly used. Might also be good to change the name of the field.

Reply:
Fixed.